### PR TITLE
✨ Make CloudInit and Sysprep raw and inlined fields pointers

### PR DESCRIPTION
**Pull request by bryanv**
This makes it easier to determine which method is being used, and in the case of CloudInit, not display the inlined config when it is otherwise not being used due to a required non-empty field.

```release-note
NONE
```

### CODE REVIEW

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go [hunk 1]
```diff
@@ -26,7 +25,7 @@ func BootstrapSysPrep(
 	var data string
 	key := "unattend"
 
-	if equality.Semantic.DeepEqual(sysPrepSpec.Sysprep, sysprep.Sysprep{}) {
+	if !reflect.ValueOf(sysPrepSpec.RawSysprep).IsZero() {
```
Comment (akutz) [inline pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go:28]: Why is this not just checking is not nil?

File: webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go [hunk 2]
```diff
@@ -226,9 +222,7 @@ func (v validator) validateBootstrap(
 		}
 
 		if lib.IsWindowsSysprepFSSEnabled() {
-			hasSysPrep := !equality.Semantic.DeepEqual(sysPrep.Sysprep, sysprep.Sysprep{})
-			hasRawSysPrep := !equality.Semantic.DeepEqual(sysPrep.RawSysprep, corev1.SecretKeySelector{})
-			if hasSysPrep && hasRawSysPrep {
+			if sysPrep.Sysprep != nil && !reflect.ValueOf(sysPrep.RawSysprep).IsZero() {
```
Comment (akutz) [inline webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go:225]: Isn't RawSysprep by address? Why the reflection?

**Review by akutz — COMMENTED** [review]
I had two small questions. If they are resolved, it looks good to me.

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go (hunk 1 above)
Comment (akutz) [reply pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go:28]: I see what happened. I expected you to make `RawCloudConfig` and `RawSysprep` by address as well.

File: webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go (hunk 2 above)
Comment (akutz) [reply webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go:225]: See previous comment.

File: pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go (hunk 1 above)
Comment (bryanv) [reply pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap_sysprep.go]: Sure. They're all pointers now.

**Review by sreyasn — APPROVED** [review]

**Pull request merged**